### PR TITLE
[air] Remove checkpoint user guide and update key concepts and docstring

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?

Removes the 'checkpoint' user guide, moves information either to
docstring or to the key concepts page.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/air/checkpoint.py [hunk 1]
```diff
@@ -42,22 +42,65 @@
 class Checkpoint:
     """Ray AIR Checkpoint.
 
-    This implementation provides methods to translate between
-    different checkpoint storage locations: Local storage, external storage
-    (e.g. cloud storage), and data dict representations.
+    An AIR Checkpoint are a common interface for accessing models across
```
Comment (ericl) [inline python/ray/air/checkpoint.py:45]:
```suggestion
    Checkpoints are a common interface for accessing models across
```

**Comment by pcmoritz** [comment]
So I think showing how our checkpointing interacts with mlflow is very useful. The example we had was bad and not showing that, but we should probably have a good example that shows how to do it (e.g. convert our checkpoints to mlflow or load checkpoints from mlflow to our format). If that is not already covered.

**Comment by richardliaw** [comment]
@pcmoritz RE: mlflow integration we don't have a good flow there right now and are planning to improve that flow in 2.1. This change here is only focused on reorganizing existing information.

File: doc/source/ray-air/key-concepts.rst [hunk 2]
```diff
@@ -37,8 +37,6 @@ See the documentation on :ref:`Trainers <air-trainer-ref>`.
     :start-after: __air_trainer_start__
     :end-before: __air_trainer_end__
 
-
-
 Trainer objects will produce a :ref:`Result <air-results-ref>` object after calling ``.fit()``.  These objects will contain training metrics as long as checkpoints to retrieve the best model.
 
 .. literalinclude:: doc_code/air_key_concepts.py
```
Comment (dmatrix) [inline doc/source/ray-air/key-concepts.rst:42]: Let's use present tense for actions which are true. More forceful.

"Trainer objects produce a :ref:`Result <air-results-ref>` object after calling ``.fit()``. These objects contain training metrics as checkpoints to retrieve the best model.

**Comment by dmatrix** [comment]
I'm not sure if Session as concept belongs here as a high-level concept. Sessions are low-level implementation details, and users of AIR API will never create a `Session` object per se. These objects, as I understand (can correct me so), are implicitly created by the Trainers and Tuners and accessible in their training code they'll write. 

Better to put that in the user guide. IMO.

**Review by dmatrix — APPROVED** [review]

**Pull request merged**